client/jetpack-onboarding/steps/summary.jsx

```diff
@@ -97,10 +104,21 @@ class JetpackOnboardingSummaryStep extends React.PureComponent {
 	}
 }
 
-export default connect( ( state, { siteId } ) => {
+export default connect( ( state, { siteId, steps } ) => {
 	const tasks = compact( [] );
+	const stepsCompleted = reduce(
+		steps,
+		// eslint-disable-next-line wpcalypso/redux-no-bound-selectors
+		( result, stepName ) => {
+			result[ stepName ] = isJetpackOnboardingStepCompleted( state, siteId, stepName );
+			return result;
+		},
+		{}
+	);
+
```

Comment:
Instead of using reduce here, what about `map`ping `stepName`s to `bool`s that indicated completion?

Reply:
Not sure I follow you here - could you please elaborate?

Reply:
I'm suggesting making `stepsCompleted` an object that would look roughly like this:

```js
{
	[ SITE_TITLE ]: true,
	[ SITE_TYPE ]: true,
	[ HOMEPAGE ]: false,
	[ CONTACT_FORM ]: false,
	[ BUSINESS_ADDRESS ]: true,
	[ WOOCOMMERCE ]: false,
}
```

or even

```js
{
	[ SITE_TITLE ]: {
		label: translate( 'Site Title & Description' )
		completed: true
	},
	[ SITE_TYPE ]:{ /* ... */ },
	/* etc */
}
```
so that we could just iterate over that object here https://github.com/Automattic/wp-calypso/pull/21440/files#diff-6a32ed70daadfc8e8e9262e1a6a3f38aR31 and have all required information contained in it. (It's a pattern that I've used elsewhere -- essentially adding an attribute to a pre-existing object-of-objects that indicates something I need to represent in the UI.)

Reply:
Well, I think the titles wouldn't belong here (they're currently defined in a constant, which makes more sense to me).

So going with the first approach sounds good, but that's exactly what the `reduce()` achieves right now. Also, the `map()` would result in having an array, not an object?

Reply:
> So going with the first approach sounds good, but that's exactly what the `reduce()` achieves right now. Also, the `map()` would result in having an array, not an object?

Oh, is it? I thought it was only producing an array that contained the completed steps.

> Also, the `map()` would result in having an array, not an object?

Lodash's `mapValues` then or whatever it takes 😄

Reply:
> Well, I think the titles wouldn't belong here (they're currently defined in a constant, which makes more sense to me).

Might be a matter of taste and data architecture -- I tend to prefer to define *one* object that holds all the (immutable) information instead of many, e.g. 

```js
export const JETPACK_ONBOARDING_STEPS = {
	'site-title': {
		label: translate( 'Site Title & Description' ),
		component: <JetpackOnboardingSiteTitleStep />,
	},
	'site-type': { /*...*/ },
	/* etc */
}
```

instead of https://github.com/Automattic/wp-calypso/blob/6fbe72fc51eaa87e389261e6614d5a7b894488cf/client/jetpack-onboarding/constants.js#L20-L56.

I will then use my 'master object' e.g. in a `connect()` to add contextual attributes that I need in the UI, such as that `completed` bool. That way, I'll pretty much always just `map` the same object to React components -- sometimes with more information added.

Reply:
> Lodash's mapValues then or whatever it takes

Sad we can't `map` at all, because we need to convert the values of the input array to keys in the output object. So, we'll have to stick with `reduce`...

> Might be a matter of taste and data architecture -- I tend to prefer to define one object that holds all the (immutable) information instead of many, e.g.

Totally agree with this one, and will fix in a subsequent PR.